code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,36 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("- no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		log_and_message_admins("has generated a DeepmaintRooms map.")
+		break
+
+	return
+
+
+/datum/admins/proc/throw_into_deepmaint(var/mob/M)
+	set category = "Fun"
+	set desc = "Throws the target into Deepmaint."
+	set name = "Send to DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("tried to throw into Deepmaint, but no Deepmaint Z-levels are currently registered!")
+		return
+
+	var/atom/movable/target = M
```

Comment:
Any reason not to typecast this in the proc parameters instead of reassigning like this? I don't _think_ the admin datum type checks when calling the proc, so would a `istype(M)` be required in place of the `isnull` on the next line?

Reply:
IIRC I did this to extend this to all atoms later; it was just annoying to have this verb pop up for all atoms, for time being.